### PR TITLE
[BIA-1163] On AMT asmt_StudentAssessmentFact view and EPP_CandidateDim view are returning StudentUSI column

### PR DESCRIPTION
**Pull request by jegasape**
- On AMT asmt_StudentAssessmentFact view and EPP_CandidateDim view are returning StudentUSI column

### CODE REVIEW

File: src/EdFi.AnalyticsMiddleTier.Tests/TestCases/StudentAssessmentFact/MSSQL/v_3_3/0001_StudentAssessmentFact_should_match_column_dictionary.xml [hunk 1]
```diff
@@ -43,6 +43,10 @@
         <ColumnName>StudentAssessmentIdentifier</ColumnName>
         <DataType>nvarchar</DataType>
     </Result>
+   <Result>
+        <ColumnName>StudentUniqueId</ColumnName>
```
Comment (DavidJGapCR) [inline src/EdFi.AnalyticsMiddleTier.Tests/TestCases/StudentAssessmentFact/MSSQL/v_3_3/0001_StudentAssessmentFact_should_match_column_dictionary.xml:47]: The new column is StudentKey

Comment (DavidJGapCR) [reply src/EdFi.AnalyticsMiddleTier.Tests/TestCases/StudentAssessmentFact/MSSQL/v_3_3/0001_StudentAssessmentFact_should_match_column_dictionary.xml:47]: Additionally, you changed this file, which is correct. But you forgot to change the same file for DataStandard 3.1 and 3.2.

**Comment by DavidJGapCR** [comment]
Since we added a new field on the `EPP_CandidateDim` view, we need to add a new test to check that the value returned by this field is correct. It should be similar to 
`\src\EdFi.AnalyticsMiddleTier.Tests\TestCases\CandidateDim\c1000042_StudentUSI_should_be_971.xml`
but, instead of checking for `StudentUSI`, it should check the new field. 
Take a look at this file:
`\src\EdFi.AnalyticsMiddleTier.Tests\Dimensions\When_querying_the_CandidateDim_view.cs`
This is where we map the queries to the tests for this specific view

**Comment by DavidJGapCR** [comment]
It looks like your commit is not signed.

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/Asmt/PostgreSQL/0024-View-StudentAssessmentFact-Alter.sql [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		'-', StudentAssessmentStudentObjectiveAssessmentPerformanceLevel.PerformanceLevelDescriptorId,
+		'-', Student.StudentUniqueId,
+		'-', StudentSchoolAssociation.SchoolId,
+		'-', TO_CHAR(StudentSchoolAssociation.EntryDate, 'yyyymmdd')
+	) AS StudentAssessmentFactKey,
+	CONCAT(
+		StudentAssessment.AssessmentIdentifier, 
+		'-', StudentAssessment.Namespace, 
+		'-', StudentAssessment.StudentAssessmentIdentifier, 
+		'-', Student.StudentUniqueId,
```
Comment (DavidJGapCR) [inline src/EdFi.AnalyticsMiddleTier.DataStandard32/Asmt/PostgreSQL/0024-View-StudentAssessmentFact-Alter.sql:27]: There is a syntax error here. You have to remove the comma.

File: src/EdFi.AnalyticsMiddleTier.Tests/TestCases/StudentAssessmentFact/MSSQL/v_3_3/0001_StudentAssessmentFact_should_match_column_dictionary.xml (hunk 1 above)
Comment (jegasape) [reply src/EdFi.AnalyticsMiddleTier.Tests/TestCases/StudentAssessmentFact/MSSQL/v_3_3/0001_StudentAssessmentFact_should_match_column_dictionary.xml]: Done.

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/Asmt/PostgreSQL/0024-View-StudentAssessmentFact-Alter.sql (hunk 2 above)
Comment (jegasape) [reply src/EdFi.AnalyticsMiddleTier.DataStandard32/Asmt/PostgreSQL/0024-View-StudentAssessmentFact-Alter.sql]: Good catch, Done!

File: src/EdFi.AnalyticsMiddleTier.Tests/Classes/StudentAssessmentFact.cs [hunk 3]
```diff
@@ -19,6 +19,7 @@ public class StudentAssessmentFact
         public string Namespace { get; set; }
         public string StudentAssessmentIdentifier { get; set; }
         public string StudentUSI { get; set; }
+        public string StudentKey { get; set; }
```
Comment (DavidJGapCR) [inline src/EdFi.AnalyticsMiddleTier.Tests/Classes/StudentAssessmentFact.cs:22]: We need to add this new field on the `CandidateDim` class as well.

Comment (jegasape) [reply src/EdFi.AnalyticsMiddleTier.Tests/Classes/StudentAssessmentFact.cs:22]: Done.

**Review by DavidJGapCR — APPROVED** [review]

**Pull request merged**